mobile/mysterium/wireguard_connection_setup.go

```diff
@@ -196,6 +200,13 @@ func (w *wireguardDeviceImpl) Start(privateKey string, config wireguard.ServiceC
 		return errors.Wrap(err, "could not create tunnel device")
 	}
 
+	oldDevice := w.device
+	defer func() {
+		if oldDevice != nil {
+			oldDevice.Close()
```

Comment:
I'm confused about this: why is the old device closed at the end, and only if it exists? Could it more direct to just close it as soon as the new one is created and assigned?

Reply:
It does exactly the same. Closes the old device once new created and configured.